backend/doc/domain/resource_lookup.py

```diff
@@ -852,187 +793,237 @@ def maybe_correct_book_name(
+    elif len(repo_components) == 2:
+        # if resource_type in usfm_resource_types:
+        #     logger.debug("FUBAR")  # DEBUG This case happened
+        #     # Get book code from USFM file name and then lookup name in English book names
+        #     usfm_files = find_usfm_files(resource_filepath)
+        #     for usfm_file in usfm_files:
+        #         book_code = Path(usfm_file).stem.lower().split("-")[1]
+        #         book_codes_and_names.append((book_code, book_names[book_code]))
+        if resource_type in ["tn", "tq"]:
+            # Get book code from TN and TQ repo book sub-directory
+            # names and use to lookup in English book names
+            subdirs = [
+                file
+                for file in scandir(resource_filepath)
+                if file.is_dir() and file.name in book_names
+            ]
+            for subdir in subdirs:
+                book_codes_and_names.append(
+                    (
+                        subdir.name.lower(),
+                        book_names[subdir.name.lower()],
+                    )
+                )
+    return book_codes_and_names
+
+
+def get_book_names_from_usfm_metadata(
+    resource_filepath: str,
+    lang_code: str,
+    resource_type: str,
+) -> dict[str, str]:
+    """
+    Book names obtained from USFM frontmatter/metadata may or may not
+    be localized, it depends on the translation work done for language
+    lang_code.
+    """
+    book_codes_and_names_localized: dict[str, str] = {}
+    usfm_files = parsing.find_usfm_files(resource_filepath)
+    for usfm_file in usfm_files:
+        usfm_file_components = Path(usfm_file).stem.lower().split("-")
+        book_code = usfm_file_components[1]
+        usfm = read_file(usfm_file) if usfm_file else ""
+        frontmatter, _, _ = parsing.split_usfm_by_chapters(
+            lang_code, resource_type, book_code, usfm
+        )
+        localized_book_name = parsing.maybe_localized_book_name(frontmatter)
+        # localized_book_name = maybe_correct_book_name(lang_code, localized_book_name)
+        book_codes_and_names_localized[book_code] = localized_book_name
+    logger.debug("book_codes_and_names_localized: %s", book_codes_and_names_localized)
+    return book_codes_and_names_localized
+
+
 @worker.app.task
 def book_codes_for_lang(
     lang_code: str,
-    resource_assets_dir: str = settings.RESOURCE_ASSETS_DIR,
-    book_names: Mapping[str, str] = BOOK_NAMES,
-    dcs_mirror_git_username: str = "DCS-Mirror",
-    usfm_resource_types: Sequence[str] = settings.USFM_RESOURCE_TYPES,
-    use_localized_book_name: bool = settings.USE_LOCALIZED_BOOK_NAME,
 ) -> Sequence[tuple[str, str]]:
     """
-    >>> from doc.domain import resource_lookup
-    >>> ();result = resource_lookup.book_codes_for_lang("pt-br");() # doctest: +ELLIPSIS
-    (...)
-    >>> result[0]
-    ('gen', 'Gênesis')
+    >>> from doc.domain.resource_lookup import book_codes_for_lang
```

Comment:
I always enjoy seeing Python doctests.  They're such an elegant way of expressing the function's contract.